functions/index.ts

```diff
@@ -0,0 +1,168 @@
+const AWS = require('aws-sdk');
+const docClient = new AWS.DynamoDB.DocumentClient();
+const table = process.env.DYNAMO_DB_TABLE;
+import { Game as Game, IHttpResponse as IHttpResponse } from "./Modules";
+const allowedRequestParameters = JSON.parse(process.env.ALLOWED_REQUEST_PARAMETERS!);
+
+exports.handler = async (event: any, context: any, callback: any) => {
+  switch (event.path) {
+    case ("/getGame"):
+      await getGame(event.queryStringParameters["gameName"]);
+      break;
+    case ("/listGames"):
+      await listGames();
+      break;
+    case("/createGame"):
+      await createGame(new Game(JSON.parse(event.body).gameName, JSON.parse(event.body).yearReleased, JSON.parse(event.body).genre, JSON.parse(event.body).console, JSON.parse(event.body).developer));
+      break;
+    case("/modifyGame"):
+      await modifyGame(new Game(JSON.parse(event.body).gameName, JSON.parse(event.body).yearReleased, JSON.parse(event.body).genre, JSON.parse(event.body).console, JSON.parse(event.body).developer));
+      break;
+    case ("/deleteGame"):
+      await deleteGame(JSON.parse(event.body).gameName);
+      break;
+    default:
+      HttpResponse({statusCode: 400, body: JSON.stringify("Unable to process request")});
+      break;
+  }
+  
+  async function getGame(gameName: string) {
+    let params = {
+      TableName: table,
+      Key: {
+        gameName: gameName
+      }
+    };
+
+    try {
+      let response = await docClient.get(params).promise();
+      if (response.Item) {
+        let game = new Game(response.Item.gameName, response.Item.yearReleased, response.Item.genre, response.Item.console, response.Item.developer);
+        HttpResponse({statusCode: 200, body: JSON.stringify(game)});
+      } else {
+        throw Error;
+     }
+    } catch (err) {
+      throw HttpResponse({statusCode: 404, body: `Unable to find game.`});
```

Comment:
In general you want to make sure that the error is because the game isn't in the DB here and wasn't because something else went wrong. Its ok to still throw a 500 when something exceptional happened that we didn't plan on.